server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -2630,18 +2630,6 @@ else if (memTable == getTabletMemory().getMemTable())
           rebuildReferencedLogs();
         }
 
-        if (numAdded > 1) {
-          // expect to add all or none
-          throw new IllegalArgumentException(
-              "Added subset of logs " + extent + " " + more + " " + currentLogs);
-        }
-
-        if (numContained > 1) {
-          // expect to contain all or none
-          throw new IllegalArgumentException(
-              "Other logs contained subset of logs " + extent + " " + more + " " + otherLogs);
-        }
-
         if (numAdded > 0 && numContained == 0) {
```

Comment:
Now that the other conditions have been removed (because they can't happen), it seems the only valid values for these are `0` and `1`. If so, this code can be dramatically simplified if we change the type to `boolean` and use assignments to set the values directly, rather than `if` blocks.